### PR TITLE
Meeting 2 Task 1

### PR DESCRIPTION
**Pull request by Massimo2511**

### CODE REVIEW

File: src/main/java/com/start/learn/human/Human.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.start.learn.human;
+
+import java.time.LocalDate;
+
+public class Human {
+    private String name;
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:6]: empty line

File: src/main/java/com/start/learn/human/Human.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        int humanID = nextHumanId;
+        nextHumanId++;
+        return humanID;
+    }
+
+    public static int getNextHumanId() {
+        return nextHumanId;
+    }
+
+    public Human(int year, int month, int day) {
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:22]: constructors are first after fileds declaration

File: src/main/java/com/start/learn/human/Human.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        dayOfBirth = LocalDate.of(year, month, day);
+
+    }
+
+    public Human(String eName, String eSurname, int yearOfBirth, int monthOfBirth, int dayOfBirth) {
+        name = eName;
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:29]: this. name = name;

File: src/main/java/com/start/learn/human/Human.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    public String getSurname() {
+        return surname;
+    }
+
+    public LocalDate getDayOfBirth() {
+        return dayOfBirth;
+    }
+
+    public void setNewNameAndSurname(Human person) {
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:46]: two fucntions, separate, and parameters to what to change

File: src/main/java/com/start/learn/human/Human.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        this.name = "Mister";
+        this.surname = "White";
+    }
+
+    public static void printHumanInfo(Human person) {
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:51]: toString() for @ Override function from Object class, so than any print result function that use object.toString() will use your implementation

**Review by iikarak — CHANGES_REQUESTED** [review]

File: src/main/java/com/start/learn/human/Human.java (hunk 1 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:6]: done

File: src/main/java/com/start/learn/human/Human.java (hunk 2 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:22]: done

File: src/main/java/com/start/learn/human/Human.java (hunk 3 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:29]: done

File: src/main/java/com/start/learn/human/Human.java (hunk 4 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:46]: done

File: src/main/java/com/start/learn/human/Human.java (hunk 5 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:51]: done

File: src/test/java/HumanMainFunction.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        System.out.println("Info about Person2 before surname change");
+        printPersonInfo(personInfo);
+        humanPersonThree.setNewSurname("Bigstock");
+        personInfo=Human.toString(humanPersonThree);
+        System.out.println("Info about Person2 after surname change");
+        printPersonInfo(personInfo);
+
+        Human humanPersonFour = new Human(1992, 12, 28);
+        personInfo=Human.toString(humanPersonFour);
+        printPersonInfo(personInfo);
```
Comment (iikarak) [inline src/test/java/HumanMainFunction.java:32]: humanPersonFour.printPersonInfo()

File: src/main/java/com/start/learn/human/Human.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    public static String toString(Human person) {
+        String generalPersonInfo;
+        if (person.name != null && person.surname != null) {
+            generalPersonInfo = String.format("Human name and surname= %s %s , BirthDay: %tD ", person.getName(), person.getSurname(), person.getDayOfBirth());
+        } else {
+            generalPersonInfo = String.format("Human ID=%s, BirthDay: %tD ", person.getHumanId(), person.getDayOfBirth());
+        }
+        return generalPersonInfo;
+    }
+    public static void printPersonInfo(String  generalPersonInfo) {
```
Comment (iikarak) [inline src/main/java/com/start/learn/human/Human.java:53]: do not pass parameter just call inside function method toString();

make as object function, not static

**Review by iikarak — CHANGES_REQUESTED** [review]

File: src/main/java/com/start/learn/human/Human.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    private String name;
+    private String surname;
+    private final LocalDate dayOfBirth;
+    private int humanId = assignId();
+    private static int nextHumanId = 1;
+
+    public Human(int year, int month, int day) {
+        this.dayOfBirth = LocalDate.of(year, month, day);
+    }
+    public Human(String Name, String Surname, int yearOfBirth, int monthOfBirth, int dayOfBirth) {
```
[the remark this answers is not in the document]
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:16]: done

File: src/main/java/com/start/learn/human/Human.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        this.name = Name;
+        this.surname = Surname;
+        this.dayOfBirth = LocalDate.of(yearOfBirth, monthOfBirth, dayOfBirth);
```
[the remark this answers is not in the document]
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:19]: done

File: src/main/java/com/start/learn/human/Human.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    public LocalDate getDayOfBirth() {
+        return dayOfBirth;
+    }
+    public void setNewName(String newFirstName) {
+        this.name = newFirstName;
+    }
+    public void setNewSurname(String newSurname) {
+        this.surname = newSurname;
+    }
+    public static String toString(Human person) {
```
[the remark this answers is not in the document]
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:44]: done

File: src/main/java/com/start/learn/human/Human.java (hunk 7 above)
Comment (Massimo2511) [reply src/main/java/com/start/learn/human/Human.java:53]: done

File: src/test/java/HumanMainFunction.java (hunk 6 above)
Comment (Massimo2511) [reply src/test/java/HumanMainFunction.java:32]: done

**Review by iikarak — APPROVED** [review]

**Pull request merged**